### PR TITLE
docs: add storybook

### LINKED ISSUE #598: Add Storybook in a way that works well with Next.js

### PR DESCRIPTION
**Pull request by mattrosno**
Closes #598 

Adds Storybook that can be ran standalone or at the same time as dev.

#### Changelog

**New**

- Stories for FrameworkIcon and asset landing page as examples

**Changed**

- Fixed FrameworkIcon prop type (incorrectly had a `status` prop)
- Web app from camelCase CSS Modules to the default, to improve source code readability and also work in Storybook

**Removed**

- Class names that had to corresponding CSS declaration blocks

#### Testing / reviewing

* `npm --workspace services/web-app run storybook` runs Storybook standalone
* `npm --workspace services/web-app run dev:all` runs Storybook and Next.js concurrently and opens browser tabs for both

1. View every page (including the MDX sandbox page) at all breakpoints to ensure this refactor didn't remove styles

### CODE REVIEW

File: services/web-app/.storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    builder: 'webpack5'
+  },
+  webpackFinal: async (config) => {
+    /**
+     * Add support for alias-imports
+     * @see https://github.com/storybookjs/storybook/issues/11989#issuecomment-715524391
+     */
+    config.resolve.alias = {
+      ...config.resolve?.alias,
+      '@': [path.resolve(__dirname, '../src/'), path.resolve(__dirname, '../')]
```
Comment (jdharvey-ibm) [inline services/web-app/.storybook/main.js:42]:
```suggestion
      '@': [path.resolve(__dirname, '..', 'src'), path.resolve(__dirname, '..')]
```

File: services/web-app/.storybook/main.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    /**
+     * Fixes font import with /
+     * @see https://github.com/storybookjs/storybook/issues/12844#issuecomment-867544160
+     */
+    config.resolve.roots = [path.resolve(__dirname, '../public'), 'node_modules']
```
Comment (jdharvey-ibm) [inline services/web-app/.storybook/main.js:49]:
```suggestion
    config.resolve.roots = [path.resolve(__dirname, '..', 'public'), 'node_modules']
```

File: services/web-app/.storybook/main.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    // Add Sass support
+    config.module.rules.push({
+      test: /\.scss$/,
+      use: ['style-loader', 'css-loader', 'sass-loader'],
+      include: path.resolve(__dirname, '../')
```
Comment (jdharvey-ibm) [inline services/web-app/.storybook/main.js:55]:
```suggestion
      include: path.resolve(__dirname, '..')
```

File: services/web-app/package.json [hunk 4]
```diff
@@ -14,11 +14,14 @@
...
     "clean": "rimraf .carbon public/.carbon .next .dev",
     "dev": "next dev",
+    "dev:all": "concurrently 'open http://localhost:3000 && next dev' 'npm run storybook'",
     "dev:clean": "npm run clean && npm run dev",
     "dev:secure": "sh scripts/create-local-certificates.sh && concurrently 'npm run dev' 'RUNNING_SECURELY=1 npm run start:proxy'",
     "service:build": "npm run build",
     "start": "concurrently 'npm run start:proxy' 'next start'",
     "start:proxy": "node proxy-server.js",
+    "storybook": "start-storybook -p 6006",
+    "build-storybook": "build-storybook -o .storybook-static",
```
[the remark this answers is not in the document]
Comment (jdharvey-ibm) [reply services/web-app/package.json:24]: minor: alphabetize

File: services/web-app/package.json [hunk 5]
```diff
@@ -59,5 +62,16 @@
...
+  },
+  "devDependencies": {
+    "@storybook/addon-actions": "^6.4.22",
+    "@storybook/addon-essentials": "^6.4.22",
+    "@storybook/addon-links": "^6.4.22",
+    "@storybook/addon-postcss": "^2.0.0",
+    "@storybook/builder-webpack5": "^6.4.22",
+    "@storybook/manager-webpack5": "^6.4.22",
+    "@storybook/react": "^6.4.22",
+    "sass-loader": "^12.6.0",
```
Comment (jdharvey-ibm) [inline services/web-app/package.json:74]: is this dependency only needed to build storybook?

File: services/web-app/.storybook/styles.module.scss [hunk 6]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright IBM Corp. 2022, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/* stylelint-disable declaration-no-important */
+/* stylelint-disable declaration-property-value-disallowed-list */
```
Comment (jdharvey-ibm) [inline services/web-app/.storybook/styles.module.scss:9]: Should we put comments next to these indicating why they're needed in this file?

File: services/web-app/.gitignore [hunk 7]
```diff
@@ -38,6 +38,8 @@ yarn-error.log*
...
 
 # certificates
-
 /certificates/localhost.crt
-/certificates/localhost.key
\ No newline at end of file
+/certificates/localhost.key
+
+# storybook
+.storybook-static
```
Comment (jdharvey-ibm) [inline services/web-app/.gitignore:45]: minor: file is missing a new line at the end

File: services/web-app/package.json (hunk 5 above)
Comment (mattrosno) [reply services/web-app/package.json:74]: Yes, `sass-loader` needed for Storybook webpack.

**Comment by mattrosno** [comment]
@jdharvey-ibm I just pushed a commit with your recommended changes.

**Review by alisonjoseph — APPROVED** [review]
Looked through all the pages and mdx components and found some unrelated bugs (opened 2 issues), but everything here looks good! 🔥

**Comment by andreancardona** [comment]
Tried to do a pretty thorough regression test - found some unrelated bugs (opening issues). Looks good!! 🎉

**Review by andreancardona — APPROVED** [review]

**Review by francinelucca — APPROVED** [review]
LGTM

**Pull request merged**